flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testframe/utils/MetricQueryer.java

```diff
@@ -0,0 +1,163 @@
+            throws ConfigurationException {
+        restClient = new RestClient(configuration, executor);
+    }
+
+    public JobDetailsInfo getJobDetails(TestEnvironment.Endpoint endpoint, JobID jobId)
+            throws Exception {
+        String jmAddress = endpoint.getAddress();
+        int jmPort = endpoint.getPort();
+
+        final JobMessageParameters params = new JobMessageParameters();
+        params.jobPathParameter.resolve(jobId);
+
+        return restClient
+                .sendRequest(
+                        jmAddress,
+                        jmPort,
+                        JobDetailsHeaders.getInstance(),
+                        params,
+                        EmptyRequestBody.getInstance())
+                .get(30, TimeUnit.SECONDS);
+    }
+
+    public AggregatedMetricsResponseBody getMetricList(
+            TestEnvironment.Endpoint endpoint, JobID jobId, JobVertexID vertexId) throws Exception {
+        AggregatedSubtaskMetricsParameters subtaskMetricsParameters =
+                new AggregatedSubtaskMetricsParameters();
+        Iterator<MessagePathParameter<?>> pathParams =
+                subtaskMetricsParameters.getPathParameters().iterator();
+        ((JobIDPathParameter) pathParams.next()).resolve(jobId);
+        ((JobVertexIdPathParameter) pathParams.next()).resolve(vertexId);
+        return restClient
+                .sendRequest(
+                        endpoint.getAddress(),
+                        endpoint.getPort(),
+                        AggregatedSubtaskMetricsHeaders.getInstance(),
+                        subtaskMetricsParameters,
+                        EmptyRequestBody.getInstance())
+                .get(30, TimeUnit.SECONDS);
+    }
+
+    public AggregatedMetricsResponseBody getMetrics(
+            TestEnvironment.Endpoint endpoint, JobID jobId, JobVertexID vertexId, String filters)
+            throws Exception {
+        AggregatedSubtaskMetricsParameters subtaskMetricsParameters =
+                new AggregatedSubtaskMetricsParameters();
+        Iterator<MessagePathParameter<?>> pathParams =
+                subtaskMetricsParameters.getPathParameters().iterator();
+        ((JobIDPathParameter) pathParams.next()).resolve(jobId);
+        ((JobVertexIdPathParameter) pathParams.next()).resolve(vertexId);
+        MetricsFilterParameter metricFilter =
+                (MetricsFilterParameter)
+                        subtaskMetricsParameters.getQueryParameters().iterator().next();
+        metricFilter.resolveFromString(filters);
+
+        return restClient
+                .sendRequest(
+                        endpoint.getAddress(),
+                        endpoint.getPort(),
+                        AggregatedSubtaskMetricsHeaders.getInstance(),
+                        subtaskMetricsParameters,
+                        EmptyRequestBody.getInstance())
+                .get(30, TimeUnit.SECONDS);
+    }
+
+    public Double getMetricByRestApi(
```

Comment:
Maybe `getAggregatedMetricsByRestAPI` is more accurate as the function returns `Double` type value.

Reply:
fixed